From 4d43ad4001c5196e5cde5b7c009bf57fec1b64da Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Tue, 5 Apr 2011 22:02:21 +0000
Subject: [PATCH] NET-401 FTP class uses BufferedReader for control channel,
 which does not follow the standard. Changed reader to CRLFLineReader.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@1089262 13f79535-47bb-0310-9956-ffa450edef68
---
 src/main/java/org/apache/commons/net/ftp/FTPClient.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/commons/net/ftp/FTPClient.java b/src/main/java/org/apache/commons/net/ftp/FTPClient.java
index 444db1ec0..5c3ea94f9 100644
--- a/src/main/java/org/apache/commons/net/ftp/FTPClient.java
+++ b/src/main/java/org/apache/commons/net/ftp/FTPClient.java
@@ -44,6 +44,7 @@
 import org.apache.commons.net.ftp.parser.FTPFileEntryParserFactory;
 import org.apache.commons.net.ftp.parser.MLSxEntryParser;
 import org.apache.commons.net.ftp.parser.ParserInitializationException;
+import org.apache.commons.net.io.CRLFLineReader;
 import org.apache.commons.net.io.CopyStreamAdapter;
 import org.apache.commons.net.io.CopyStreamEvent;
 import org.apache.commons.net.io.CopyStreamException;
@@ -753,7 +754,7 @@ protected void _connectAction_() throws IOException
             {
                  setControlEncoding("UTF-8");
                  _controlInput_ =
-                     new BufferedReader(new InputStreamReader(_input_, getControlEncoding()));
+                     new CRLFLineReader(new InputStreamReader(_input_, getControlEncoding()));
                  _controlOutput_ =
                     new BufferedWriter(new OutputStreamWriter(_output_, getControlEncoding()));
             }
